Source/WebAssembly/WebAssemblyBinary.cpp

```diff
@@ -1606,7 +1606,11 @@ namespace WebAssemblyBinary
 		auto segmentArenaData = outModule->arena.copyToArena(data,numDataBytes);
 		outModule->dataSegments.push_back({8,numDataBytes,segmentArenaData});
 		outModule->initialNumBytesMemory = numDataBytes + 8;
+#if INTPTR_MAX == INT32_MAX
+		outModule->maxNumBytesMemory = 0x40000000;
+#else
 		outModule->maxNumBytesMemory = 1ull << 32;
+#endif
```

Comment:
It would be best to just use 1GB here, regardless of whether the host is 32-bit. Then you can use PrintWAST to convert a binary file to text, and it will produce the same result on both 32-bit and 64-bit systems.